File: frontend-pc/apps/web-ele/src/api/monitor/login-log.ts
import { requestClient } from '#/api/request';

export interface LoginLogDto {
  id: string;
  ip: string;
  browser: string;
  os: string;
  device: string;
  browserInfo: string;
  status: boolean;
  location: string;
  message: string;
  userId: string;
  userName: string;
  realName: string;
  loginTime: string;
  userAgent: string;
}

export interface GetLoginLogsInput {
  skipCount: number;
  maxResultCount: number;
  userName?: string;
  ip?: string;
  status?: boolean;
}

export interface PagedResultDto<T> {
  totalCount: number;
  items: T[];
}

/**
 * 获取登录日志列表
 */
export async function getLoginLogListApi(params: GetLoginLogsInput) {
  return requestClient.get<PagedResultDto<LoginLogDto>>('/basic/login-log', {
    params,
  });
}

File: backend-api/framework/Cits.Domain/LoginLogs/LoginLog.cs
using Cits.Entities;

namespace Cits.LoginLogs;

public class LoginLog : Entity
{
    /// <summary>
    /// 登录IP
    /// </summary>
    public string? IP { get; set; }

    /// <summary>
    /// 浏览器
    /// </summary>
    public string? Browser { get; set; }

    /// <summary>
    /// 操作系统
    /// </summary>
    public string? Os { get; set; }

    /// <summary>
    /// 设备
    /// </summary>
    public string? Device { get; set; }

    /// <summary>
    /// 浏览器信息
    /// </summary>
    public string? BrowserInfo { get; set; }
    

    /// <summary>
    /// 登录状态（true成功，false失败）
    /// </summary>
    public bool Status { get; set; }

    /// <summary>
    /// 登录地点
    /// </summary>
    public string? Location { get; set; }

    /// <summary>
    /// 操作信息
    /// </summary>
    public string? Message { get; set; }

    /// <summary>
    /// 用户Id
    /// </summary>
    public string? UserId { get; set; }

    /// <summary>
    /// 用户名
    /// </summary>
    public string? UserName { get; set; }

    /// <summary>
    /// 姓名（昵称）
    /// </summary>
    public string? RealName { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime LoginTime { get; set; }
    
    public string? UserAgent { get; set; }
}